app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -40,38 +46,63 @@ constructor(
-      when (val userResponse = userRepository.getUser(uid)) {
-        is Resource.Success -> {
-          val user = userResponse.data!!
-          val attendeeList = user.eventsAttendeeSet.toList()
-          if (attendeeList.isNotEmpty()) {
-            when (val events = eventRepository.getEventsByIds(attendeeList)) {
-              is Resource.Success -> {
-                _uiState.value =
-                    _uiState.value.copy(
-                        eventList =
-                            EventList(
-                                events.data, events.data, _uiState.value.eventList.selectedEvent))
-              }
-              is Resource.Failure -> {
-                Log.d("EventsOverviewViewModel", "Error getting events for $uid")
-                _uiState.value =
-                    _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
-              }
-            }
-          } else {
+      userRepository.getCurrentUserId().let { userIdResource ->
+        when (userIdResource) {
+          is Resource.Success -> {
+            val uid = userIdResource.data
+            getUserEvents(uid)
+          }
+          is Resource.Failure -> {
+            Log.d(
+                "EventsOverviewViewModel",
+                "Error fetching user ID: ${userIdResource.throwable.message}")
             _uiState.value =
                 _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
           }
         }
-        is Resource.Failure -> {
-          Log.d("EventsOverviewViewModel", "Error fetching user document")
+      }
+    }
+  }
+
+  private suspend fun getUserEvents(uid: String) {
+    when (val userResponse = userRepository.getUser(uid)) {
+      is Resource.Success -> {
+        val user = userResponse.data!!
+        val attendeeList = user.eventsAttendeeSet.toList()
+        if (attendeeList.isNotEmpty()) {
+          when (val events = eventRepository.getEventsByIds(attendeeList)) {
+            is Resource.Success -> {
+              _uiState.value =
+                  _uiState.value.copy(
+                      eventList =
+                          EventList(
+                              events.data, events.data, _uiState.value.eventList.selectedEvent))
+            }
+            is Resource.Failure -> {
+              Log.d("EventsOverviewViewModel", "Error getting events for $uid")
+              _uiState.value =
+                  _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+            }
+          }
+        } else {
           _uiState.value =
               _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
         }
       }
+      is Resource.Failure -> {
+        Log.d("EventsOverviewViewModel", "Error fetching user document")
+        _uiState.value = _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+      }
+    }
+  }
+
+  fun checkUserLoginStatus() {
```

Comment:
I think this function right now looks more like updating the user's log in status. It would be nice upon Resource.Failure to Log a statement saying that no user is logged in for easy testing too.

Reply:
This function invokes getCurrentId(), which already handles logging a message if a user is not signed in. Including an additional log message here would be redundant.